rust/connlib/clients/apple/src/lib.rs

```diff
@@ -43,7 +43,7 @@ mod ffi {
             callback_handler: CallbackHandler,
         ) -> Result<WrappedSession, String>;
 
-        fn disconnect(&mut self);
+        fn disconnect(self);
```

Comment:
`disconnect` now consumes a `Session`. This ensures statically that you cannot call any functions on it after.

Reply:
I am not sure how this is represented across FFI boundaries but it would probably translate into a null pointer.

Reply:
For android there is no type safety across ffi boundaries afaik, since you're just passing pointers but idk what swift-bridge does

Reply:
At least it compiles and the apps will produce a crash-report if they use a NPE so I'd say we are good :D